cpp/models/abm/parameters.h

```diff
@@ -296,13 +296,14 @@ struct SeverityProtectionFactor {
 };
 
 /**
- * @brief Personal protective factor against high viral load. Its value is between 0 and 1.
+ * @brief Personal protective factor against high viral load, which depends on #ExposureType,
+ * #AgeGroup and #VirusVariant. Its value is between 0 and 1.
  */
 struct HighViralLoadProtectionFactor {
-    using Type = InputFunctionForProtectionLevel;
-    static auto get_default()
+    using Type = CustomIndexArray<InputFunctionForProtectionLevel, ExposureType, AgeGroup, VirusVariant>;
+    static auto get_default(AgeGroup size)
     {
-        return Type([](ScalarType /*days*/) -> ScalarType {
+        return Type({ExposureType::Count, size, VirusVariant::Count}, [](ScalarType /*days*/) -> ScalarType {
             return 0;
         });
```

Comment:
```suggestion
        return Type({ExposureType::Count, size, VirusVariant::Count}, TimeSeriesFunctor<ScalarType>());
```

As a heads up, the InputFunctionForProtectionLevel aka std::function was deleted in the serialization branch, so this has to be updated.